packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -1340,12 +1340,32 @@ class ProxyDomain extends Domain {
   Future<String> connect(Map<String, dynamic> args) async {
     final int targetPort = _getIntArg(args, 'port', required: true);
     final String id = 'portForwarder_${targetPort}_${_id++}';
-    final Socket socket = await Socket.connect('127.0.0.1', targetPort);
+
+    Socket socket;
+
+    try {
+      socket = await Socket.connect(InternetAddress.loopbackIPv4, targetPort);
+    } on SocketException {
+      // If connecting to ipv4 loopback fails, try ipv6.
+      socket = await Socket.connect(InternetAddress.loopbackIPv6, targetPort);
+    }
```

Comment:
You can also wrap the IPv6 attempt in a `try {} catch {}`.

```dart
    Socket socket;

    try {
      socket = await Socket.connect(InternetAddress.loopbackIPv4, targetPort);
    } on SocketException (e, s) {
      globals.logger.printTrace(...);
    }
    // If connecting to ipv4 loopback fails, try ipv6.
    try {
      socket ?= await Socket.connect(InternetAddress.loopbackIPv6, targetPort);
    } on SocketException (e, s) {
      globals.logger.printError(...);
    }
```

Reply:
Done